zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -216,54 +218,68 @@ CREATE TABLE orchard_received_note_spends (
     UNIQUE (orchard_received_note_id, transaction_id)
 )";
 
-/// Stores the current UTXO set for the wallet, as well as any transparent outputs
-/// previously observed by the wallet.
+/// Stores the transparent outputs received by the wallet.
 ///
 /// Originally this table only stored the current UTXO set (as of latest refresh), and the
 /// table was cleared prior to loading in the latest UTXO set. We now upsert instead of
 /// insert into the database, meaning that spent outputs are left in the database. This
-/// makes it similar to the `*_received_notes` tables in that it can store history, but
-/// has several downsides:
-/// - The table has incomplete contents for recovered-from-seed wallets.
-/// - The table can have inconsistent contents for seeds loaded into multiple wallets
+/// makes it similar to the `*_received_notes` tables in that it can store history.
+/// Depending upon how transparent TXOs for the wallet are discovered, the following
+/// may be true:
+/// - The table may have incomplete contents for recovered-from-seed wallets.
+/// - The table may have inconsistent contents for seeds loaded into multiple wallets
 ///   simultaneously.
-/// - The wallet's transparent balance can be incorrect prior to "transaction enhancement"
+/// - The wallet's transparent balance may be incorrect prior to "transaction enhancement"
 ///   (downloading the full transaction containing the transparent output spend).
-pub(super) const TABLE_UTXOS: &str = r#"
-CREATE TABLE "utxos" (
+///
+/// ### Columns:
+/// - `id`: Primary key
+/// - `transaction_id`: Reference to the transaction in which this TXO was created
+/// - `output_index`: The output index of this TXO in the transaction referred to by `transaction_id`
+/// - `account_id`: The account that controls spend authority for this TXO
+/// - `address`: The address to which this TXO was sent
```

Comment:
We don't store the address a shielded note was received to in the `*_received_notes` tables directly, because we don't need to: all we need to know is the recipient account and the note plaintext, and we can spend.

For transparent outputs, due to how our "transparent viewing keys" work, the specific address a transparent output was sent to is relevant in order to derive its corresponding spending key. However, we also already have that information in the `script` column (the address is extracted from it), and thus the seemingly-obvious reason for this column to exist is for joins. Where do we do these joins?

Another question: is `address` the right thing to store here semantically, or should we instead be storing the `(scope, address_index)` tuple that can be used to derive the address from the corresponding transparent IVK? We should be determining this at "transparent scan" time.

Reply:
I don't have time at the moment to try to figure out what the right thing is to do here. We've had the address here all along, I don't see a reason to change it right now.

Reply:
Okay, the reason that we store the address explicitly is simply that we query against it - `get_unspent_transparent_outputs` takes the address as an argument. This is because when shielding we want to avoid shielding from multiple transparent addresses in a single transaction, because that would link them.